pkg/server/plugin/keymanager/gcpkms/gcpkms.go

```diff
@@ -1103,13 +1120,41 @@ func parseAndValidateConfig(c string) (*Config, error) {
 		return nil, status.Error(codes.InvalidArgument, "configuration is missing the key ring")
 	}
 
-	if config.KeyMetadataFile == "" {
-		return nil, status.Error(codes.InvalidArgument, "configuration is missing server ID file path")
+	if config.KeyMetadataFile == "" && config.KeyIdentifierFile == "" && config.KeyIdentifierValue == "" {
+		return nil, status.Error(codes.InvalidArgument, "configuration requires server id or server id file path")
+	}
+	if (config.KeyMetadataFile != "" || config.KeyIdentifierFile != "") && config.KeyIdentifierValue != "" {
+		return nil, status.Error(codes.InvalidArgument, "configuration must not contain both server id and server id file path")
+	}
+	if config.KeyMetadataFile != "" && config.KeyIdentifierFile != "" {
+		return nil, status.Error(codes.InvalidArgument, "configuration must not contain both 'key_identifier_file' and deprecated 'key_metadata_file'")
+	}
+	if config.KeyIdentifierValue != "" {
+		if !validateCharacters(config.KeyIdentifierValue) {
+			return nil, status.Error(codes.InvalidArgument, "Key identifier must contain only alphanumeric characters, underscores (_), and dashes (-)")
+		}
+		if !unicode.IsLetter(rune(config.KeyIdentifierValue[0])) {
+			return nil, status.Error(codes.InvalidArgument, "Key identifier must start with a letter character")
+		}
```

Comment:
This restriction is only for the key of a label, not the value. As an example, UUIDs start with a number, and that should continue to be allowed.